Added New Term Entry HTML: Tags: `<q>` tag

Antonio-Riccelli opened the pull request:
<!--
👋 Hi, thanks for sending a PR to Codecademy Docs! 💖
Please fill out all fields below to ensure your PR is reviewed quickly.

**IMPORTANT**

If you would like to receive credit for your contribution to an entry, make sure to have your Codecademy user profile linked to your GitHub account:

1. Go to your Codecademy dashboard.
2. Click on your profile image in the top-right and then choose "Profile".
3. Then click "Edit Profile".
4. In the "GitHub Username" field, add your username (without the @).
5. Click "Save Changes"

Of course, you can opt not to do this and be listed as an "Anonymous contributor", instead. :)
-->

### Description

<!-- Please write a summary of the change, such as which topic(s) and file(s) that you have edited or created. Please also include relevant motivation and context: -->

I have added an entry for the `<q>` tag to the HTML section.
This is in the "content > html > tags" folder.

### Type of Change

<!--- Please delete or cross off the bullet point(s) that are irrelevant to this PR: -->

- Adding a new entry

### Checklist

<!-- Please check ALL the boxes: -->

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.

<!--
Having trouble with the PR checker? Here are some common issues and resolutions:

- verify_formatting is failing
  - run `yarn format path/to/markdown/file.md` or `yarn format:all` and commit the results
- verify_lint is failing
  - same as above
  - if verify_lint is still failing, running `yarn lint` locally should let you know what needs to be changed by hand
- test is failing
  - ensure any new markdown files have a `Title` and `Description` defined in their metadata
  - ensure any new markdown files only contain alphanumerics and dashes in their file names and have the same name as their parent directory
  - if that looks ok, running `yarn test` locally should let you know what the issue is
-->

KTom101 commented on content/html/concepts/tags/terms/q/q.md:
```diff
@@ -0,0 +1,39 @@
+---
+
+The `<q>` tag is used to represent a brief inline quotation.
+It accepts a `cite` attribute which will link to a source, providing further context for the quotation.
```
```suggestion
It accepts a `cite` attribute which will link to a source that provides further context for the quotation.
```

Antonio-Riccelli replied:
Corrected.

KTom101 commented on content/html/concepts/tags/terms/q/q.md:
```diff
@@ -0,0 +1,39 @@
+  The World Wide Web Consortium's idea of the Web includes
+  <q cite="https://www.w3.org/Consortium/mission"
+    >participation, sharing knowledge, and thereby building trust on a global
+    scale.</q
```
```suggestion
    scale.</q>
```

KTom101 commented on content/html/concepts/tags/terms/q/q.md:
```diff
@@ -0,0 +1,39 @@
+  <q cite="https://www.w3.org/Consortium/mission"
+    >participation, sharing knowledge, and thereby building trust on a global
+    scale.</q
+  >
```
```suggestion
```

KTom101 commented on content/html/concepts/tags/terms/q/q.md:
```diff
@@ -0,0 +1,39 @@
+<!-- The text between the q tags will be surrounded by quotation marks -->
+<p>
+  The World Wide Web Consortium's idea of the Web includes
+  <q cite="https://www.w3.org/Consortium/mission"
```
```suggestion
  <q cite="https://www.w3.org/Consortium/mission">
```

KTom101 commented on content/html/concepts/tags/terms/q/q.md:
```diff
@@ -0,0 +1,39 @@
+<p>
+  The World Wide Web Consortium's idea of the Web includes
+  <q cite="https://www.w3.org/Consortium/mission"
+    >participation, sharing knowledge, and thereby building trust on a global
```
```suggestion
    participation, sharing knowledge, and thereby building trust on a global
```

KTom101 commented on content/html/concepts/tags/terms/q/q.md:
````diff
@@ -0,0 +1,39 @@
+<q><!-- brief quotation goes inside the tag --></q>
+```
+
+The `<q>` tag requires an opening and closing tag wrapped around a line of text. By default, the browser will render this enclosed in quotation marks.
````
```suggestion
The `<q>` tag requires an opening and closing tag wrapped around a line of text. The browser will automatically enclose the text in quotation marks.
```

Antonio-Riccelli replied:
Corrected.

KTom101 commented on content/html/concepts/tags/terms/q/q.md:
```diff
@@ -0,0 +1,39 @@
+
+The `<q>` tag requires an opening and closing tag wrapped around a line of text. By default, the browser will render this enclosed in quotation marks.
+
+Its `cite` attribute should be a valid URL.
```
```suggestion
The `cite` attribute should be a valid URL.
```

KTom101 reviewed (commented):
@Antonio-Riccelli Just finished the 1st review for this entry. Please let us know when you're done making changes. Thanks!

Antonio-Riccelli commented:
Hi @KTom101, 
I've implemented the suggested corrections, but now the request doesn't pass the "format:verify" test. What should I do? 
Thank you

SSwiniarski commented:
@Antonio-Riccelli, failing that test means there's some formatting issue it doesn't like. (Usually an extra space or linefeed, which can be difficult to spot.) If you run "yarn format" on the file, and commit the changes, it will tend to fix these issues.

@KTom101, once you approve the changes, I can go ahead with review 2.

Dusch4593 commented:
> @Antonio-Riccelli, failing that test means there's some formatting issue it doesn't like. (Usually an extra space or linefeed, which can be difficult to spot.) If you run "yarn format" on the file, and commit the changes, it will tend to fix these issues.

If you make changes locally, on a branch of your forked repo, there's a script that should run whenever you `git add` and `git commit` your changes.

It's usually pretty good about catching those spaces and linefeeds (though I noticed the other day that Prettier doesn't seem to work with trailing spaces of code embedded in snippets or the markdown itself).

KTom101 reviewed (approved):
@SSwiniarski You can move on with review 2.

SSwiniarski commented:
@Antonio-Riccelli, I still see several of @KTom101's suggestions unresolved. Can you address them so I can do review two?

Antonio-Riccelli commented:
Hi @SSwiniarski - working on it, I should push the corrections shortly.

Dusch4593 commented:
@Antonio-Riccelli Hmmm let me have a look real quick and see. 😄

Dusch4593 commented:
@Antonio-Riccelli Try adding a space between the text and the angle-brackets.

~~`<q cite="https://www.w3.org/Consortium/mission">participation ...`~~ -> `<q cite="https://www.w3.org/Consortium/mission"> participation ...`

Then run `yarn format` again. It should look better, but let me know what you think. 😄

Antonio-Riccelli commented:
@Dusch4593 Thank you. It doesn't break the tag now, but it's failing the test...

Dusch4593 commented:
Did you re-add/commit/push the format from your local branch?

Antonio-Riccelli commented:
Yes, I ran the script, then re-added, committed and pushed.  
Should I have done it differently?

Antonio-Riccelli commented:
Hi @Dusch4593, 
I've tried again. Let me know if I'm doing it correctly, please.
  
![3](https://user-images.githubusercontent.com/39813213/155094589-23eff4cb-1901-43e0-8100-07d7803b47df.gif)

Dusch4593 commented:
@Antonio-Riccelli looks like it did! Nice work! (btw those recorded GIFs are super helpful, thank you! 😃)

Dusch4593 commented on content/html/concepts/tags/terms/q/q.md:
```diff
@@ -0,0 +1,40 @@
+It accepts a `cite` attribute which will link to a source that provides further context for the quotation.
+
+## Syntax
+
```
```suggestion

The `<q>` tag requires an opening and closing tag wrapped around a line of text. The browser will automatically enclose the text in quotation marks:

```

Antonio-Riccelli replied:
Hey @Dusch4593 - apologies for missing this one. Before I commit and to ensure I understand correctly, should the final result look like this: 

![image](https://user-images.githubusercontent.com/39813213/156812728-aa648561-5726-4dff-b829-0d187641e0ac.png)

Dusch4593 replied:
Yep, that's right! 👍

Dusch4593 commented on content/html/concepts/tags/terms/q/q.md:
````diff
@@ -0,0 +1,40 @@
+## Syntax
+
+```html
+<q><!-- brief quotation goes inside the tag --></q>
````
```suggestion
<q cite=""><!-- brief quotation goes inside the tag --></q>
```

Dusch4593 commented on content/html/concepts/tags/terms/q/q.md:
````diff
@@ -0,0 +1,40 @@
+<q><!-- brief quotation goes inside the tag --></q>
+```
+
+The `<q>` tag requires an opening and closing tag wrapped around a line of text. The browser will automatically enclose the text in quotation marks.
+
+The `cite` attribute should be a valid URL.
````
```suggestion
It accepts a `cite` attribute that links to a valid URL that provides further context for the quotation.
```

Dusch4593 commented on content/html/concepts/tags/terms/q/q.md:
```diff
@@ -0,0 +1,40 @@
+  - 'paths/front-end-engineer-career-path'
+---
+
+The `<q>` tag is used to represent a brief inline quotation.
+It accepts a `cite` attribute which will link to a source that provides further context for the quotation.
```
```suggestion
The `<q>` tag is used to represent a brief inline quotation, as opposed the [`<blockquote>` tag](https://www.codecademy.com/resources/docs/html/tags/blockquote) being used for longer ones.

```

Dusch4593 commented on content/html/concepts/tags/terms/q/q.md:
````diff
@@ -0,0 +1,40 @@
+  </q>
+  One can't help but agree with it!
+</p>
+```
````
Could we include an image of what the rendered text would look like on a browser? (preferably .jpg or .png)

Antonio-Riccelli replied:
Hi @Dusch4593, where I should store the image?

Dusch4593 replied:
You can save it either as a `.jpg` or `.png` file and save it in the folder for your PR branch (`terms/q/`). And then you add it with the usual ![...](...) markdown image syntax.

Antonio-Riccelli replied:
Thank you! Addressing these changes today.

Dusch4593 reviewed (changes requested):
Apologies for the delay, @Antonio-Riccelli. I just left some comments/suggestion for you.

Antonio-Riccelli commented:
Thank you, I'll address these today. @Dusch4593

Antonio-Riccelli commented:
Hi @Dusch4593, 
Please let me know if I've implemented suggestions correctly. 
I think that test "Push/ci (test)" is failing because of the .jpg image I've added. Is that in the right folder?

Dusch4593 commented:
> Hi @Dusch4593, Please let me know if I've implemented suggestions correctly. I think that test "Push/ci (test)" is failing because of the .jpg image I've added. Is that in the right folder?

Hey @Antonio-Riccelli! 

First of all, thank you for uploading this awesome image! It looks great! 👍 

Secondly, I apologize for misinforming you about where to put this image! 🤭 Could we actually move it to [the `media/` folder in the root directory of this project](https://github.com/Codecademy/docs/tree/main/media)? If you're not comfortable doing this, however, I'm more than happy to make the change. 😃

Afterwards, that one check should pass.

yangc95 commented on content/html/concepts/tags/terms/q/q.md:
```diff
@@ -0,0 +1,39 @@
+  - 'paths/front-end-engineer-career-path'
+---
+
+The `<q>` tag is used to represent a brief inline quotation, as opposed the [`<blockquote>` tag](https://www.codecademy.com/resources/docs/html/tags/blockquote) being used for longer ones.
```
```suggestion
The `<q>` tag is used to represent a brief inline quotation, as opposed to the [`<blockquote>` tag](https://www.codecademy.com/resources/docs/html/tags/blockquote) being used for longer ones.
```

yangc95 commented on content/html/concepts/tags/terms/q/q.md:
````diff
@@ -0,0 +1,39 @@
+## Syntax
+
+```html
+<q cite=""><!-- brief quotation goes inside the tag --></q>
````
For these comments, I'm thinking that we should capitalize the first word and end with a punctuation~

yangc95 replied:
```suggestion
<q cite=""><!-- Brief quotation goes inside the tag. --></q>
```

yangc95 reviewed (commented):
@Antonio-Riccelli while you're working on moving the image, I thought I would include another review for this entry.

Thanks again for your contribution 🙏 Please tag me or @Dusch4593 after you've resolved all the comments/suggestions ☺️

Antonio-Riccelli commented:
Hi @yangc95,
I've just implemented the changes and moved the image. It seems to have worked. Let me know if further amendments are required :)

Dusch4593 reviewed (approved):
@Antonio-Riccelli Things LGTM on my end! 👍 There was one more suggested change I had left outstanding if you wanted to check it out and resolve it?

Dusch4593 commented:
Thanks @Antonio-Riccelli! @yangc95 anything left to check before we approve/merge?

yangc95 reviewed (approved):
LGTM!

Dusch4593 commented:
Congrats @Antonio-Riccelli your contribution is now live on Docs! 

https://www.codecademy.com/resources/docs/html/tags/q

Dusch4593 commented:
@Antonio-Riccelli Hmm looks like that image didn't render the way we expected:

<img width="687" alt="Screen Shot 2022-03-04 at 12 58 23 PM" src="https://user-images.githubusercontent.com/23144845/156816910-71f835cb-985b-46cc-b9f6-2185d84ea8d8.png">

I'll work on a fix right now! 👍

Antonio-Riccelli commented:
@Dusch4593 Let me know if I have to change anything 😯 It seemed to render fine locally.

Dusch4593 commented:
We just merged [a patch PR](https://github.com/Codecademy/docs/pull/591) to hopefully fix the problem. Let us know if it's still rendering ok on your side. 😄
